src/overlays/actors/ovl_En_Mt_tag/z_en_mt_tag.c

```diff
@@ -35,42 +40,474 @@ const ActorInit En_Mt_tag_InitVars = {
+                                         &perpendicularPointsX[i], &perpendicularPointsZ[i]);
+        if (highestCurrentCheckpoint < currentCheckpoints[i]) {
+            highestCurrentCheckpoint = currentCheckpoints[i];
+        }
+    }
+
+    for (i = 1; i < ARRAY_COUNT(this->raceGorons) + 1; i++) {
+        rg = this->raceGorons[i - 1];
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Mt_tag/func_809CF350.s")
+        // Because of the bug described in EnMttag_GetCurrentCheckpoint, these values may not be initialized.
+        //! @bug When initialized, this check is pointless because upcomingCheckpoint is always 0 or higher.
+        if ((upcomingCheckpoints[i] != -1) && (upcomingCheckpoints[0] != -1)) {
+            rg->numCheckpointsAheadOfPlayer = (upcomingCheckpoints[i] - upcomingCheckpoints[0]);
+        } else {
+            rg->numCheckpointsAheadOfPlayer = 0;
+        }
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Mt_tag/func_809CF394.s")
+    if ((currentCheckpoints[0] > 0) && (currentCheckpoints[0] < highestCurrentCheckpoint) &&
+        (player->actor.bgCheckFlags & 1) && ((highestCurrentCheckpoint - currentCheckpoints[0]) >= 24)) {
+        ret = true;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Mt_tag/func_809CF444.s")
+    return ret;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Mt_tag/func_809CF4EC.s")
+/**
+ * Exits the race and returns the player back to "normal" gameplay.
+ * Whether the player won or lost the race is determined by arg1 and nextTransition.
+ */
+s32 EnMttag_ExitRace(GlobalContext* globalCtx, s32 arg1, s32 nextTransition) {
+    CUR_FORM_EQUIP(EQUIP_SLOT_B) = ITEM_SWORD_KOKIRI;
+    globalCtx->nextEntranceIndex = 0xD020;
+    if ((gSaveContext.weekEventReg[33] & 0x80)) {
+        // Spring
+        gSaveContext.nextCutsceneIndex = 0xFFF0;
+    } else {
+        // Winter
+        gSaveContext.nextCutsceneIndex = 0;
+    }
+
+    globalCtx->sceneLoadFlag = 0x14;
+    globalCtx->unk_1887F = arg1;
+    gSaveContext.nextTransition = nextTransition;
+    func_801477B4(globalCtx);
+    return 1;
+}
+
+/**
+ * Displays the text which says that the player has made a false start.
+ */
+void EnMttag_ShowFalseStartMessage(EnMttag* this, GlobalContext* globalCtx) {
+    gSaveContext.unk_3DD0[4] = 0;
+    func_801518B0(globalCtx, 0xE95, NULL); // An entrant made a false start
+    func_800B7298(globalCtx, &this->actor, 7);
+    Audio_QueueSeqCmd(0x101400FF);
+    this->actionFunc = EnMttag_PotentiallyRestartRace;
+}
+
+/**
+ * Displays the text from the Goron Elder's child which tells the player that
+ * they probably can't win the race.
+ */
+void EnMttag_ShowCantWinMessage(EnMttag* this, GlobalContext* globalCtx) {
+    func_801518B0(globalCtx, 0xE97, NULL); // You can't win now...
+    func_800B7298(globalCtx, &this->actor, 7);
+    this->actionFunc = EnMttag_HandleCantWinChoice;
+}
```

Comment:
Nothing special happens, unfortunately. You just win like normal. This makes sense, since this actor just reverts back to running `EnMttag_Race` as soon as you say you want to continue racing.